src/webui/api/searchcontroller.cpp

```diff
@@ -59,6 +57,34 @@ namespace
         if (activeSearches.remove(id))
             session->setData(ACTIVE_SEARCHES, QVariant::fromValue(activeSearches));
     }
+
+    /**
+    * Returns the search categories in JSON format.
+    *
+    * The return value is an array of dictionaries.
+    * The dictionary keys are:
+    *   - "id"
+    *   - "name"
+    */
+    QJsonArray getPluginCategories(QStringList categories)
+    {
+        QJsonArray categoriesInfo;
+
+        categoriesInfo << QJsonObject {
```

Comment:
you can initialize it:
```c++
QJsonArray categoriesInfo {
    {QLatin1String("id"), "all"},
    {QLatin1String("name"), SearchPluginManager::categoryFullName("all")}
};
```

Reply:
Unfortunately it's actually a little uglier than that
```cpp
        QJsonArray categoriesInfo {QJsonObject {
            {QLatin1String("id"), "all"},
            {QLatin1String("name"), SearchPluginManager::categoryFullName("all")}
        }};
```